### DISCUSSION TITLE
Move "Listeners" always to the last posotions

### DISCUSSION THREAD

**Discussion by andy1239** [jamulussoftware/jamulus#1639] [Ideas]
When listeners join session a lot of channels are moved... Could they just be at the end in general?

**Comment by hoffie** [comment]
Hrm, are you already sorting your faders by Instrument? If not, this should provide at least some more convience. If you do, I suppose you suggest changing that sort order and special-casing Listener to be last?

**Reply by andy1239** [reply]
I am. And if Jamulus' language is set to german, all "Zuhörer" are at the end. 
Bit "Listener" starts with l ;-)

**Reply by jp8** [reply]
Sorting by instrument should (in my view) result in the same sort order no matter what the language.  It should not be by alphabetical order of instrument names in the user's language.  Rather it should be something similar to the order a conductor would put his musicians in... flutes at the front, drummers at the back, and so on.

And yes, maybe listeners always at the right.

**Reply by dcorson-ticino-com** [reply]
> it should be something similar to the order a conductor would put his musicians in...

There is no standard.  I play in multiple orchestras, none have the same arrangement of players, some even changing from concert to concert.  Even just for the string section there are two "standards".

It will be hard to find something more intuitive than alphabetic in the users language.

**Reply by jp8** [reply]
I think at least the conductor's score order is largely standardized, granted with a few minor differences among musical genres.

If we went with something like:  wind, brass, guitar, keys, bass, drums, voices, strings, listeners, (with instruments from high to low within each section) we would be close to what's needed, I think.

**Reply by pljones** [reply]
You'd need somewhere for things like piano, xylophone, harp, etc, too.

The easiest would be to simplify the instruments list to those sections, plus "other"...

But any such arrangement is probably going to raise more disagreement than concord.

**Comment by gilgongo** [comment]
See also: https://github.com/jamulussoftware/jamulus/discussions/1646

**Reply by reinhardwh** [reply]
Own mixer to be first in the list sounds useful to me.

**Comment by ghost** [comment]
If there are several sort criteria, they should be prioritised by the user
in the user settings.
eg. sort A third, sort B first, sort C none, sort D Nth sort E none

No sort check box should be on the main menu.
eg. the DIR command in MSDOS can be configured perfectly in
autoexec.bat or config.sys to perfect the directory listing.
(it appears modern operating systems have lost this ability)

**Comment by gilgongo** [comment]
> When listeners join session a lot of channels are moved... Could they just be at the end in general?

This reminds me that one of my explorations into a new UI for Jamulus had an option to join a server as a listener or convert yourself to one once joined. Listeners were then shown simply as a number (which you could click to get a list of names), and marking yourself as a listener defaulted you to being in muted state (overridable).
